src/oncall/scheduler/default.py

```diff
@@ -0,0 +1,320 @@
+        else:
+            return None
+
+    def get_roster_user_ids(self, roster_id, cursor):
+        cursor.execute('''
+            SELECT `roster_user`.`user_id` FROM `roster_user`
+            JOIN `user` ON `user`.`id` = `roster_user`.`user_id`
+            WHERE `roster_user`.`in_rotation` = 1 AND `roster_user`.`roster_id` = %r
+                AND `user`.`active` = TRUE''', roster_id)
+        return [r['user_id'] for r in cursor]
+
+    def get_busy_user_by_event_range(self, user_ids, team_id, events, cursor):
+        ''' Find which users have overlapping events for the same team in this time range'''
+        range_check = []
+        for e in events:
+            range_check.append('(%r < `end` AND `start` < %r)' % (e['start'], e['end']))
+
+        query = '''
+                SELECT DISTINCT `user_id` FROM `event`
+                WHERE `user_id` in %%s AND team_id = %%s AND (%s)
+                ''' % ' OR '.join(range_check)
+
+        cursor.execute(query, (user_ids, team_id))
+        return [r['user_id'] for r in cursor.fetchall()]
+
+    def find_least_active_user_id_by_team(self, user_ids, team_id, start_time, role_id, cursor):
+        '''
+        Of the people who have been oncall before, finds those who haven't been oncall for the longest. Start
+        time refers to the start time of the event being created, so we don't accidentally look at future
+        events when determining who was oncall in the past. Done on a per-role basis, so we don't take manager
+        or vacation shifts into account
+        '''
+        cursor.execute('''
+            SELECT `user_id`, MAX(`end`) AS `last_end` FROM `event`
+            WHERE `team_id` = %s AND `user_id` IN %s AND `end` <= %s
+            AND `role_id` = %s
+            GROUP BY `user_id`
+            ''', (team_id, user_ids, start_time, role_id))
+        if cursor.rowcount != 0:
+            # Grab user id with lowest last scheduled time
+            return min(cursor.fetchall(), key=lambda x: x['last_end'])['user_id']
+        else:
+            return None
+
+    def find_new_user_in_roster(self, roster_id, team_id, start_time, role_id, cursor):
+        '''
+        Return roster users who haven't been scheduled for any event on this team's calendar for this schedule's role.
+        Ignores events from other teams.
+        '''
+        query = '''
+            SELECT DISTINCT `user`.`id` FROM `roster_user`
+            JOIN `user` ON `user`.`id` = `roster_user`.`user_id` AND `roster_user`.`roster_id` = %s
+            LEFT JOIN `event` ON `event`.`user_id` = `user`.`id` AND `event`.`team_id` = %s AND `event`.`end` <= %s
+                AND `event`.`role_id` = %s
+            WHERE `roster_user`.`in_rotation` = 1 AND `event`.`id` IS NULL
+        '''
+        cursor.execute(query, (roster_id, team_id, start_time, role_id))
+        if cursor.rowcount != 0:
+            logger.debug('Found new guy')
+        return set(row['id'] for row in cursor)
```

Comment:
kind of a nitpick, but this can be a set comprehensions as we're 2.7+

    return {row['id'] for row in cursor}